sway-core/src/semantic_analysis/ast_node/declaration.rs

```diff
@@ -54,7 +57,11 @@ impl CopyTypes for TypedDeclaration {
             ConstantDeclaration(ref mut const_decl) => const_decl.copy_types(type_mapping),
             FunctionDeclaration(ref mut fn_decl) => fn_decl.copy_types(type_mapping),
             TraitDeclaration(ref mut trait_decl) => trait_decl.copy_types(type_mapping),
-            StructDeclaration(ref mut struct_decl) => struct_decl.copy_types(type_mapping),
+            StructDeclaration(ref decl_id) => {
+                let mut struct_decl = de_get_struct(decl_id.clone(), &self.span()).unwrap();
```

Comment:
I know this PR is still blocked for other reasons, but when https://github.com/FuelLabs/sway/pull/2682 goes in, will you remove this? It should allow us to remove the use of `.unwrap()` inside of implementations of `CopyTypes`.